subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/repositories/AuthenticationSupporter.java

```diff
@@ -89,6 +89,11 @@ public void credentials(Action<? super PasswordCredentials> action) {
         action.execute(getCredentials(credentialsType));
     }
 
+    @Override
+    public void credentials(Class<? extends Credentials> credentialsType) {
+        throw new UnsupportedOperationException();
```

Comment:
This class implements the public `AuthenticationSupported` interface. `AbstractAuthenticationSupportedRepository` also implements `AuthenticationSupported` and is the entry point when using repositories. It delegates the credentials related method to this one. In this case, it delegates to the next method in this class, adding the identity supplier.

Perhaps this class should not implement `AuthenticationSupported` - it is only used as an implementation detail in repositories.